### PR TITLE
Add groundboost indicator for defrag

### PR DESCRIPTION
**Pull request by PeenScreeker**
Closes momentum-mod/game/issues/2035 (convar implemented in red repo)

This PR adds a groundboost feature for defrag. The indicator shows when the player has the knockback flag (from weapon or teleporter) and changes state to show a successful crash landing to extend the defrag timer to max duration.

Currently intended to be implemented for testing and feedback - no settings UI included in this PR. The feature as implemented is only available through the console.

### Checks

-   [x] **I have thoroughly tested all of the code I have modified/added/removed to ensure something else did not break**
-   [x] I have followed [semantic commit messages](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) e.g. `feat: Add foo`, `chore: Update bar`, etc...
-   [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
-   [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review
-   [x] All changes requested in review have been `fixup`ed into my original commits.
-   [x] Fully tokenized all my strings (no hardcoded English strings!!) and supplied [bulk JSON strings](https://docs.momentum-mod.org/guide/localization/#bulk-adding-terms) below

### CODE REVIEW

File: styles/controls/progress-bar.scss [hunk 1]
```diff
@@ -38,3 +38,17 @@
...
+	box-shadow: $progressbar-shadow;
+
+	.CircularProgressBarFG {
+		//background-color: gradient(radial, 50% 50%, 0% 0%, 50%, 50%, from($light-blue), to($blue));
+	}
+
+	.CircularProgressBarBG {
+		background-color: $progressbar-background;
+	}
+}
```
Comment (tsa96) [inline styles/controls/progress-bar.scss:54]: Interesting seeing use of this thing at last, but it's not actually being used anywhere in XML? Definitely seems like something that belongs in a separate commit at least

File: styles/hud/ground-boost.scss [hunk 2]
```diff
@@ -0,0 +1,45 @@
+@use '../config' as *;
+
+.groundboost {
+	&__container {
+		height: 100px;
+		width: 100px;
+		margin: 6px 0px 6px 0;
+
+		transition-property: opacity;
+		transition-duration: 0.5s;
```
[the remark this answers is not in the document]
Comment (PeenScreeker) [reply styles/hud/ground-boost.scss:10]: yea, linear is the default but I don't mind calling it out explicitly.

File: styles/controls/progress-bar.scss (hunk 1 above)
Comment (PeenScreeker) [reply styles/controls/progress-bar.scss:54]: Yea, I should have removed this. I started with this implementation and scrapped it. [progress-bar.scss](https://github.com/momentum-mod/panorama/pull/80/files/ce709da79e7d4ad3477bd6b321f6e509e04d7f47#diff-1f7f07fdfb6116dcd40601a66a312d733cfd4b8bad3fbef4ec0d003ea3f74eba) shouldn't even be modified.

File: scripts/hud/ground-boost.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	NONE: 0,
+	LANDING: 1 << 0,
+	KNOCKBACK: 1 << 1,
+	WATERJUMP: 1 << 2
+};
+
+const COLOR_CLASS = {
+	KB: 'groundboost__meter--knockback',
+	GB: 'groundboost__meter--groundboost'
+};
```
[the remark this answers is not in the document]
Comment (tsa96) [reply scripts/hud/ground-boost.js:13]: Still needs addressing

File: styles/hud/ground-boost.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (tsa96) [reply styles/hud/ground-boost.scss:10]: Still needs addressing (though tbf it doesn't really matter). Though, have you tried non-linear functions? For fades `ease-in-out` or `ease-out` tends to look much better

File: styles/hud/ground-boost.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
+@use '../config' as *;
+
+.groundboost {
+	&__container {
+		height: 100px;
+		width: 100px;
+		margin: 6px 0px 6px 0;
```
[the remark this answers is not in the document]
Comment (tsa96) [reply styles/hud/ground-boost.scss]: Most CSS devs seem to prefer `0` to `0px`, doesn't really matter though.

**Review by tsa96 — APPROVED** [review]

**Review by Gocnak — APPROVED** [review]
